### PR TITLE
feature/JC19-359-display-detail

### PR DESCRIPTION
**Pull request by daoduong-saritasa**
Display anime detail

### CODE REVIEW

File: apps/vanilla/src/pages/detail/detail.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { AnimeService } from '../../services/animeService';
+import { AuthService } from '../../services/authService';
+
+window.addEventListener('load', async() => {
+  await AuthService.navigateByAuthorization();
+  await Navbar.renderNavbar();
+  const animeId = localStorage.getItem(CLICKED_ANIME);
+  assertNonNull(animeId);
+  const animeDetail = await AnimeService.getAnimeDetail(animeId);
+  renderTitle(animeDetail.englishTitle ? animeDetail.englishTitle : animeDetail.japaneseTitle);
```
Comment (gevorg-gevorgyan-saritasa) [inline apps/vanilla/src/pages/detail/detail.ts:15]:
```suggestion
  renderTitle(animeDetail.englishTitle || animeDetail.japaneseTitle);
```

File: apps/vanilla/src/pages/login/index.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <link rel="stylesheet" href="./login.css" />
+    <title>Login</title>
+  </head>
+  <body class="default-layout">
+    <h1 class="title">Login</h1>
+    <form class="row s8 form">
+      <div class="col s8">
+          <label> Email
+            <input type="email" name="email" placeholder="Email" required autocomplete="on">
+          </label>
```
Comment (gevorg-gevorgyan-saritasa) [inline apps/vanilla/src/pages/login/index.html:15]:
```suggestion
          <label> 
            Email
            <input type="email" name="email" placeholder="Email" required autocomplete="on">
          </label>
```

File: apps/vanilla/src/scripts/animeTable.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+/** Handle click on anime rows. */
+export function handleAnimeClick(): void {
+  const animeTableRows = document.querySelectorAll<HTMLTableRowElement>('tbody tr');
+  assertNonNull(animeTableRows);
+  animeTableRows.forEach(animeRow => {
+    animeRow.addEventListener('click', () => {
+      const animeId = animeRow.getAttribute('data-id');
+      assertNonNull(animeId);
+      localStorage.setItem(CLICKED_ANIME, animeId);
```
Comment (gevorg-gevorgyan-saritasa) [inline apps/vanilla/src/scripts/animeTable.ts:61]: Consider using your `StorageService` to work with `localStorage`

File: apps/vanilla/src/services/animeService.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    assertNonNull(video);
+    modalWrapperOverlay.addEventListener('click', () => {
+    video.setAttribute('src', '');
+    modalWrapper.classList.add('hidden');
+  });
+  }
+
+  /**
+   * Open full size image.
+   * @param imageURL Youtube trailer Id.
```
Comment (gevorg-gevorgyan-saritasa) [inline apps/vanilla/src/services/animeService.ts:108]: Update docs please

File: apps/vanilla/src/pages/index.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  };
+  const anime = await AnimeService.getAnime(paginationConfig);
+  renderTable(anime);
+  renderSortOptions();
+  initSearch();
+};
+
+/** Reset the table state. */
+export function resetTableState(): void {
+  localStorage.setItem(PaginationLocalStorage.active, FIRST_PAGE.toString());
```
Comment (gevorg-gevorgyan-saritasa) [inline apps/vanilla/src/pages/index.ts:34]: Use `StorageService` here

File: apps/vanilla/src/services/storageService.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+   */
+  export async function remove(key: string): Promise<void> {
+    localStorage.removeItem(key);
+  }
+
+  /**
+   * Set token to storage.
+   * @param token The token to set.
+   */
+  export async function setToken(token: TokenDto): Promise<void> {
```
Comment (gevorg-gevorgyan-saritasa) [inline apps/vanilla/src/services/storageService.ts:43]: Why you need to make all this functions `async`?

**Review by gevorg-gevorgyan-saritasa — CHANGES_REQUESTED** [review]

File: libs/core/models/anime/animeDetail.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import { Type, Status } from '../../dtos/anime/anime.dto';
+
+import { OmitImmerable, Immerable } from '../immerable';
+import { DateRange } from '../dateRange';
+
+import { Genre } from './genre';
+import { Studio } from './studio';
+
+/** Anime Detail. */
+export class AnimeDetail extends Immerable {
```
Comment (timofei-maximov-saritasa) [inline libs/core/models/anime/animeDetail.ts:11]: This is extended `anime` type, let's extend it from `anime` to reduce code duplication

File: libs/core/models/auth/register.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+/** Login user model. */
+export interface Register{
```
Comment (timofei-maximov-saritasa) [inline libs/core/models/auth/register.ts:2]:
```suggestion
export interface Register {
```

File: apps/vanilla/src/services/animeService.ts [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    return PaginationMapper.fromDto(animeResponse.data, animeDto => AnimeMapper.fromDto(animeDto));
+  }
+
+  /**
+   * Get anime detail from the server.
+   * @param id ID of the anime.
+   */
+  export async function getAnimeDetail(id: string): Promise<AnimeDetail> {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const animeDetailResponse = await api.get(`anime/anime/${id}/`);
```
Comment (timofei-maximov-saritasa) [inline apps/vanilla/src/services/animeService.ts:36]: `get` method accepts generic

File: apps/vanilla/src/pages/detail/detail.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+import { AnimeService } from '../../services/animeService';
+import { AuthService } from '../../services/authService';
+import { StorageService } from '../../services/storageService';
+
+window.addEventListener('load', async() => {
+  await AuthService.navigateByAuthorization();
+  await Navbar.render();
+  const animeId = await StorageService.get<string>(CLICKED_ANIME);
+  assertNonNull(animeId);
```
Comment (timofei-maximov-saritasa) [inline apps/vanilla/src/pages/detail/detail.ts:15]: What if user will use direct link to alime details? You can use query params to retrieve anime id

**Pull request merged**